src/js/modules/users/selectors.js

```diff
@@ -12,67 +12,65 @@ const getRequestedIllustratorSlug = (state, props) =>
 const getRequestedPhotographerSlug = (state, props) =>
   props.match.params.photographer_slug;
 
-const getContributorRoleId = state => state.users.CONTRIBUTOR_ROLE_ID;
-const getIllustratorRoleId = state => state.users.ILLUSTRATOR_ROLE_ID;
-const getPhotographerRoleId = state => state.users.PHOTOGRAPHER_ROLE_ID;
-
 const getRoleFromProps = (state, props) => props.role;
 
 export const getContributorFromSlug = createSelector(
-  [getUsers, getUserRoles, getRequestedContributorSlug, getContributorRoleId],
-  (users, userRoles, requestedContributorSlug, CONTRIBUTOR_ROLE_ID) => {
+  [getUsers, getUserRoles, getRequestedContributorSlug, getRoles],
+  (users, userRoles, requestedContributorSlug, roles) => {
     const user = Object.values(users).find(user => {
       return user.slug === requestedContributorSlug;
     });
     if (
       userRoles.find(userRole => {
         return (
-          userRole.userId === user.id && userRole.roleId === CONTRIBUTOR_ROLE_ID
+          userRole.userId === user.id &&
+          roles[userRole.roleId].title === "Contributor"
         );
       })
     ) {
       return user;
     }
-    appHistory.push("/the-uncompleted-404-page");
+    appHistory.push("/the-uncompleted-404-page?ref=contributors");
```

Comment:
Add a TODO to figure out how to NotFound. I don't want to add 404 to EVERY single component here.